terraform/ec2-instances-creator/main.tf

```diff
@@ -0,0 +1,55 @@
+locals {
+  # Available fields in each element:
+  #  {
+  #    "fbVersion": "2.0.7",
+  #    "ec2User": "centos",
+  #    "osDistro": "centos",
+  #    "osVersion": 9,
+  #    "arch": "x86_64",
+  #    "ami": "ami-08c92aec9ccf0e1e9",
+  #    ...
+  #  }
+  instance_matrix = jsondecode(file(var.instance_matrix_file))
+
+  aws_vpc_subnet = "subnet-06240c469195932cf"
+  ec2_instances_security_group = "sg-0c4318be91bbe3cba"
+  ec2_instance_profile = "logging-e2e-testing-ec2-ssm-instance-profile"
+
+  # See: https://docs.aws.amazon.com/systems-manager/latest/userguide/ssm-agent-status-and-restart.html
+  # TODO Verify for Ubuntu: it seems Ubuntu >18 uses snap, but verify if systemctl also starts SSM agent correctly
+  linux_user_data_boot_script_for_ssm = <<EOF
+#!/bin/bash
+systemctl enable amazon-ssm-agent
+systemctl start amazon-ssm-agent
+EOF
```

Comment:
Right now we're just using this one, but I've left here some breadcrumbs for Windows and Ubuntu for when we need them (in the near future).